src/python/pants/base/specs.py

```diff
@@ -341,6 +353,12 @@ def to_spec_string(self) -> str:
     return self.glob
 
 
+@dataclass(frozen=True)
+class FilesystemResolvedGlobSpec(FilesystemGlobSpec, FilesystemResolvedSpec):
+  """A spec with resolved globs, e.g. `*.py` may resolve to `('f1.py', 'f2.py', '__init__.py')`."""
+  resolved_files: Tuple[str, ...]
```

Comment:
FYI how this works: https://docs.python.org/3/library/dataclasses.html#inheritance

tl;dr: inheriting the dataclass `FilesystemGlobSpec` means that we will automatically get the field `glob: str`. We add an additional field `resolved_files: Tuple[str, ...]`.